scripts/coverage.sh

```diff
@@ -42,7 +45,19 @@ RUST_LOG=solana=trace _ cargo +$rust_nightly test --target-dir target/cov "${pac
 
 echo "--- grcov"
 
-_ grcov target/cov/debug/deps/ > target/cov/lcov-full.info
+# Create a clean room dir only with updated gcda/gcno files for this run
+data_dir=target/cov/tmp
+rm -rf "$data_dir"
+mkdir -p "$data_dir"
+
+find target/cov -name \*.gcda -newer "$timing_file" |
```

Comment:
I'm curious why you didn't go for a simple for loop:
```bash
for gcda_file in $(find target/cov -name \*.gcda -newer "$timing_file"); do
  declare gcno_file="${gcda_file%.gcda}.gcno"
  ln -s "../../../$gcda_file" "$data_dir/$(basename "$gcda_file")"
  ln -s "../../../$gcno_file" "$data_dir/$(basename "$gcno_file")"
done
```

Reply:
No problem. :) That's a natural question. In short, I wanted to avoid another shellcheck exception. (SC2044). And among various alternative loop constructions, I ended to choose this particular one based on existing other uses of `find` in this script after some enduring time for compat. thinking. ;)

As a bonus for this round of commenting, I've made it more robust: https://github.com/solana-labs/solana/pull/7576/commits/97fe9d964d7267458b2d170c4f4559570963abe1.

Ideally, I wanted to write like this (but I can't due to the restriction of bash 3+) based on [this](https://github.com/koalaman/shellcheck/wiki/SC2044#if-find-is-just-matching-globs-recursively):

```patch
ryoqun@ubuqun:~/work/solana/solana$ git diff
diff --git a/scripts/coverage.sh b/scripts/coverage.sh
index 7a0df653a..3b985712f 100755
--- a/scripts/coverage.sh
+++ b/scripts/coverage.sh
@@ -51,12 +51,14 @@ mkdir -p target/cov/tmp
 
 # Can't use a simpler construct under the condition of SC2044 and bash 3
 # (macOS's default). See: https://github.com/koalaman/shellcheck/wiki/SC2044
-find target/cov -name \*.gcda -newer "$timing_file" -print0 |
-  (while IFS= read -r -d '' gcda_file; do
+(
+  shopt -s globstar nullglob
+  for gcda_file in target/cov/**/*.gcda; do
     gcno_file="${gcda_file%.gcda}.gcno"
     ln -s "../../../$gcda_file" "target/cov/tmp/$(basename "$gcda_file")"
     ln -s "../../../$gcno_file" "target/cov/tmp/$(basename "$gcno_file")"
-  done)
+  done
+)
 
 _ grcov target/cov/tmp > target/cov/lcov-full.info

```